src/jdk.compiler/share/classes/com/sun/tools/javac/code/Flags.java

```diff
@@ -73,41 +74,62 @@ public static EnumSet<Flag> asFlagSet(long flags) {
 
     /* Standard Java flags.
      */
+    @Use({FlagTarget.METHOD, FlagTarget.TYPE, FlagTarget.VARIABLE})
     public static final int PUBLIC       = 1;
+    @Use({FlagTarget.METHOD, FlagTarget.TYPE, FlagTarget.VARIABLE})
     public static final int PRIVATE      = 1<<1;
+    @Use({FlagTarget.METHOD, FlagTarget.TYPE, FlagTarget.VARIABLE})
     public static final int PROTECTED    = 1<<2;
+    @Use({FlagTarget.BLOCK, FlagTarget.METHOD, FlagTarget.TYPE, FlagTarget.VARIABLE})
     public static final int STATIC       = 1<<3;
+    @Use({FlagTarget.METHOD, FlagTarget.TYPE, FlagTarget.VARIABLE})
     public static final int FINAL        = 1<<4;
+    @Use({FlagTarget.METHOD})
     public static final int SYNCHRONIZED = 1<<5;
+    @Use({FlagTarget.VARIABLE})
     public static final int VOLATILE     = 1<<6;
+    @Use({FlagTarget.VARIABLE})
     public static final int TRANSIENT    = 1<<7;
+    @Use({FlagTarget.METHOD})
     public static final int NATIVE       = 1<<8;
+    @Use({FlagTarget.TYPE})
     public static final int INTERFACE    = 1<<9;
+    @Use({FlagTarget.METHOD, FlagTarget.TYPE})
     public static final int ABSTRACT     = 1<<10;
+    @Use({FlagTarget.TYPE})
     public static final int STRICTFP     = 1<<11;
 
     /* Flag that marks a symbol synthetic, added in classfile v49.0. */
+    @Use({FlagTarget.METHOD, FlagTarget.TYPE, FlagTarget.VARIABLE})
     public static final int SYNTHETIC    = 1<<12;
 
     /** Flag that marks attribute interfaces, added in classfile v49.0. */
+    @Use({FlagTarget.TYPE})
     public static final int ANNOTATION   = 1<<13;
 
     /** An enumeration type or an enumeration constant, added in
      *  classfile v49.0. */
+    @Use({FlagTarget.TYPE})
     public static final int ENUM         = 1<<14;
 
     /** Added in SE8, represents constructs implicitly declared in source. */
+    @Use({FlagTarget.TYPE, FlagTarget.VARIABLE})
     public static final int MANDATED     = 1<<15;
 
+    @NotFlag
     public static final int StandardFlags = 0x0fff;
 
     // Because the following access flags are overloaded with other
     // bit positions, we translate them when reading and writing class
     // files into unique bits positions: ACC_SYNTHETIC <-> SYNTHETIC,
     // for example.
+    @NotFlag
     public static final int ACC_SUPER    = 0x0020;
```

Comment:
Why didn't we add annotation on these ACC ones? After all, they _also_ overlap with previous flags -- e.g. ACC_SUPER overlaps with SYNCHRONIZED, and it works because the former goes on types and the latter on methods. So, isn't this the same trick we're pulling?